subsys/bluetooth/mesh/crypto_mtls.c

```diff
@@ -213,12 +227,124 @@ int bt_mesh_aes_cmac(const uint8_t key[16], struct bt_mesh_sg *sg,
 	return err;
 }
 
+int bt_mesh_pub_key_gen(void)
+{
+	psa_key_attributes_t key_attributes = PSA_KEY_ATTRIBUTES_INIT;
+	psa_status_t status;
+	int err = 0;
+	size_t key_len;
+
+	psa_destroy_key(key.priv_key_id);
+	key.is_ready = false;
+
+	/* Crypto settings for ECDH using the SHA256 hashing algorithm,
+	 * the secp256r1 curve
+	 */
+	psa_set_key_usage_flags(&key_attributes, PSA_KEY_USAGE_DERIVE);
+	psa_set_key_lifetime(&key_attributes, PSA_KEY_LIFETIME_VOLATILE);
+	psa_set_key_algorithm(&key_attributes, PSA_ALG_ECDH);
+	psa_set_key_type(&key_attributes, PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1));
+	psa_set_key_bits(&key_attributes, 256);
+
+	/* Generate a key pair */
+	status = psa_generate_key(&key_attributes, &key.priv_key_id);
+	if (status != PSA_SUCCESS) {
+		err = -EIO;
+		goto end;
+	}
+
+	status = psa_export_public_key(key.priv_key_id, key.public_key_be,
+				sizeof(key.public_key_be), &key_len);
+	if (status != PSA_SUCCESS) {
+		err = -EIO;
+		goto end;
+	}
+
+	if (key_len != PUB_KEY_SIZE + 1) {
+		err = -ERANGE;
+		goto end;
+	}
+
+	key.is_ready = true;
+
+end:
+	psa_reset_key_attributes(&key_attributes);
+
+	return err;
+}
+
+const uint8_t *bt_mesh_pub_key_get(void)
+{
+	return key.is_ready ? key.public_key_be + 1 : NULL;
+}
+
+BUILD_ASSERT(PSA_RAW_KEY_AGREEMENT_OUTPUT_SIZE(
+	PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1), 256) == DH_KEY_SIZE,
+	"Diffie-Hellman shared secret size should be the same in PSA and BLE Mesh");
```

Comment:
No. If this assertion happened that means BLE MESH is not compatible anymore with PSA API that crypto library implements and should be adapted.
Since there will be used couple of libraries (at least mbedTLS, TFM, Oberon) I'm pretty sure Mesh shall check it to avoid misimplementation.
The current implementation is compatible regarding PSA v1.0.1